overlord/ifacestate/helpers.go

```diff
@@ -102,15 +102,26 @@ func (m *InterfaceManager) reloadConnections(snapName string) error {
 
 func setupSnapSecurity(task *state.Task, snapInfo *snap.Info, repo *interfaces.Repository) error {
 	st := task.State()
-	var snapState snapstate.SnapState
 	snapName := snapInfo.Name()
-	if err := snapstate.Get(st, snapName, &snapState); err != nil {
-		task.Errorf("cannot get state of snap %q: %s", snapName, err)
-		return err
+
+	var devMode bool
+	// check the TaskSnapSetup for data first, then the on-disk state
+	// (no more candidate means we need to look at two places here)
+	ss, err := snapstate.TaskSnapSetup(task)
+	if err == nil {
+		devMode = ss.DevMode()
+	} else {
+		var snapState snapstate.SnapState
+		snapName := snapInfo.Name()
+		if err := snapstate.Get(st, snapName, &snapState); err != nil {
+			task.Errorf("cannot get state of snap %q: %s", snapName, err)
+		}
+		devMode = snapState.DevMode()
```

Comment:
As you pointed out in telegram this code is now wrong, because it looks for ss.DevMode() regardless what snapInfo got passed in. There is a test missing that would catch the issue.